### ISSUE TITLE
Issue when running inference /data/one-file.txt

### ISSUE THREAD

**Issue by hichem-abdellali** [jfemiani/facade-segmentation#1]
```
I1024 09:59:28.228960   100 net.cpp:248] Memory required for data: 1065139200
/usr/local/lib/python2.7/dist-packages/skimage/transform/_warps.py:84: UserWarning: The default mode, 'constant', will be changed to 'reflect' in skimage 0.15.
  warn("The default mode, 'constant', will be changed to 'reflect' in "
F1024 09:59:37.120355   100 syncedmem.cpp:51] Check failed: error == cudaSuccess (2 vs. 0)  out of memory
*** Check failure stack trace: ***
1 of 1/usr/bin/inference: line 37:   100 Aborted                 (core dumped) python -m pyfacades --plot --resume --no-use-mask --output "${OUTPUT}" --files $1
updating: output/one-file/one-file.txt (deflated 28%)
```

**Comment by jfemiani** [comment]
It looks like your GPU is out of memory.    I will look into providing a
CPU flag,  and try to document how much memory it needs. We have used this
on GTX1070, GTX1080, and K40s.

On Oct 24, 2017 6:04 AM, "Raziel" <notifications@github.com> wrote:

I1024 09:59:28.228960   100 net.cpp:248] Memory required for data: 1065139200
/usr/local/lib/python2.7/dist-packages/skimage/transform/_warps.py:84:
UserWarning: The default mode, 'constant', will be changed to
'reflect' in skimage 0.15.
  warn("The default mode, 'constant', will be changed to 'reflect' in "
F1024 09:59:37.120355   100 syncedmem.cpp:51] Check failed: error ==
cudaSuccess (2 vs. 0)  out of memory
*** Check failure stack trace: ***
1 of 1/usr/bin/inference: line 37:   100 Aborted                 (core
dumped) python -m pyfacades --plot --resume --no-use-mask --output
"${OUTPUT}" --files $1
updating: output/one-file/one-file.txt (deflated 28%)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/jfemiani/facade-segmentation/issues/1>, or mute the
thread

**Comment by djidan10** [comment]
Thanks for replying, I would also ask you kindly if you can explain the results made by this code briefly or how to interpret it (I was able to run this code on an Amazon instance with 12 GiB GPU Memory, my machine has 2 Gib GPU Memory)  Thanks

**Comment by jfemiani** [comment]
This is repository is part of a project I did with a colleague at UCL to reconstruct 3D buildings. I output a .yml file that includes the estimated bounding boxes of a variety of facade elements captured from google street-view images.  The output folder also includes a variety of diagnostic images, including a probability mask for each image that shows the probability that each pixel is of a certain type (e.g. window, door, etc.).     The .yml file is not really documented (at least not yet) but I hope it is decipherable anyhow. 

It may be useful to try and run `jupyter notebook` from within the docker container, and look at the `scripts/process.ipynb`.   I added a bit to the README.md explaining how to launch the notebook server, hopefully seeing the notebook might give a bit more insight into how to understand the result.

**Comment by djidan10** [comment]
Hi, 
Thanks for replying, I read the paper and it's clear now I have just one more additional question 
What the meta.txt file is contained and how did you produce the mask for the input?
Thanks :)

**Comment by zzqstar** [comment]
@djidan10  Hi, Could you send a image of facade-segmentation that you pulled earlier to me ?  I pull the image later and found that there are some problem with the image ...

**Comment by jfemiani** [comment]
Here -- I tested it and I live-streamed to YouTube.
The audio is bad, but I show the whole process I think.

I found that the dockerhub `jfemiani/segnet-facade:cuda8-cudnn3` tag needed
a 'git pull' to bring it up to date with the git repo.  Both the `latest`
and `cuda8-cudnn3` images  are now up to date with github.

Ideally it would automatically update, but since I cannot build the images
without a GPU setup I have to manually commit the images if I ever edit the
git repo.

https://www.youtube.com/watch?v=lKyFnZ6ThTo

**Comment by zzqstar** [comment]
@jfemiani  Thank you for your work ! In dockerhup, tags information of  jfemiani/segnet-facade show that the images were updated 8 days ago recently, not few hours ago ...

**Comment by zzqstar** [comment]
Thank you very much!  A more question involved that the gpu devices 
can't be detected even run as nvidia-docker ("nvidia-smi" comman not found in the container ),
many methods having been tried but the problem still exists. Is there a rigidity 
requirement of  the gpu version(like Tesla that you are using) or gpu memory(the gpu can't 
be detected with a gpu memory less than certain threhold)? Thank you!

**Comment by jfemiani** [comment]
Not that I am aware of.  You should be able to use the cpu if the gpu is
giving you trouble.

**Comment by jfemiani** [comment]
@zzqstar Were you able to use the GPU? One thing I have found is that sadly there is some coupling between the nvidia drivers and libraries on the host and in the docker container you are running.

**Comment by zzqstar** [comment]
Yeah, @jfemiani thank you very much!  I cannot use the GPU now. The problem  is not solved yet. What's the coupling bewteen the nvidia drivers and libraries no the host?
